library/x509_create.c

```diff
@@ -112,6 +113,19 @@ int mbedtls_x509_string_to_names( mbedtls_asn1_named_data **head, const char *na
             s = c + 1;
             in_tag = 0;
             d = data;
+            if( *s == '\"')
+            {
+                in_quot = 1;
+                s++;
+                c += 2;
+            }
+        }
+
+        if( in_quot && *c == '\"' && c != end )
+        {
+            c++;
+            s = c + 1;
+            in_quot = 0;
```

Comment:
Is it possible for this code to overread from the buffer pointed to by s and c? That is, the check at the begining of the loop (`c <= end`) does not guarantee anything other than "we can read the character at c". For example in line 116 the new code reads the location at c+1 without checking that `c+1 <= end`, but I am not familiar enough with the code to know that there exists a path through the code that causes such errors. What do you think?

I think that tests like this are on the border of fuzzing, but perhaps we could create a couple of tests that check corner cases? For example, an unterminated quote or a name that is empty after the equal and is at the end of the string.

Reply:
I see your point. I'll see how I can fix it.
In addition, I think the whole code should probably change from `if ( c != end )` to `if( c <= end )`.
What do you think?
I'll add some more corner case checks as well